### ISSUE TITLE
Unknown database adapter `"mysql2"` found in config/database.yml, use Gemfile.local to load your own database gems

### ISSUE THREAD

**Issue by thejulian** [docker-library/redmine#353]
Since 5.1.5 the container won't start. I guess the reason are the changes in the [Gemfile](https://github.com/redmine/redmine/blob/5.1.4/Gemfile):58 for the adapters.
5.1.4 works just fine with the same docker run command.

**docker run command**
```
docker run --name redmine -e REDMINE_DB_MYSQL=redminedb -e REDMINE_DB_USERNAME=redmine -e REDMINE_DB_PASSWORD=password -e REDMINE_DB_DATABASE=redmine --link redminedb redmine:5.1.5
``` 

**Error log**
``` 
Unknown database adapter `"mysql2"` found in config/database.yml, use Gemfile.local to load your own database gems
Your Gemfile lists the gem puma (>= 0) more than once.
You should probably keep only one of them.
Remove any duplicate entries and specify the gem only once.
While it's not a problem now, it could cause errors if you change the version of one of them later.
The Gemfile's dependencies are satisfied
Unknown database adapter `"mysql2"` found in config/database.yml, use Gemfile.local to load your own database gems
Unknown database adapter `"mysql2"` found in config/database.yml, use Gemfile.local to load your own database gems
rake aborted!
LoadError: Error loading the 'mysql2' Active Record adapter. Missing a gem it depends on? mysql2 is not part of the bundle. Add it to your Gemfile. (LoadError)
/usr/local/bundle/gems/activerecord-6.1.7.10/lib/active_record/connection_adapters/mysql2_adapter.rb:6:in `<top (required)>'
```

**Comment by hectorm** [comment]
This is due to a recent upstream change regarding the parsing of the `database.yml` file: https://www.redmine.org/issues/41749#note-14

**Comment by drexlma** [comment]
Workaround, remove ": 

Change
```
 production:
  adapter: "mysql2"
```

to 

```
production:
  adapter: mysql2
```

**Comment by LeeThompson** [comment]
I've tried mounting my own external `config/database.yml` but it's not reading it.

**Comment by MadTiago** [comment]
Having the same issue when trying to update to v6.

**Comment by LeeThompson** [comment]
I spent some time trying to get a corrected yml mounted for the container and never got it to read it.     This is major fumble for Redmine.    I know it's due to a "up the chain" issue but it's like it wasn't even tested.    

I'm not sure what database changes there have been made so I'm not in a rush to downgrade but currently it can't run so I guess nothing vs something?

Really disappointed in Redmine for letting this hit us.

**Comment by thejulian** [comment]
The breaking changes happened with the modifications in Redmine 5.1.5 and 6.0.2. 
Redmine 5.1.4 and 6.0.1 are working.

**Comment by hagaygo** [comment]
> @LeeThompson The breaking changes happened with the modifications in Redmine 5.1.5 and 6.0.2. Redmine 5.1.4 and 6.0.1 are working.

Same here , error when upgrading from 6.0.1 to 6.0.2

**Comment by ishikawa999** [comment]
This issue was fixed on December 19, 2024, so I think it will be resolved in the next version.

https://www.redmine.org/issues/42013
